oqsprov/oqsprov_keys.c

```diff
@@ -628,17 +622,22 @@ static int oqsx_hybsig_init(int bit_security, OQSX_EVP_CTX *evp_ctx,
 
 static const int oqshybkem_init_ecp(char *tls_name, OQSX_EVP_CTX *evp_ctx,
                                     OSSL_LIB_CTX *libctx) {
-    int ret = 1;
-    int idx = 0;
+    int ret = 0;
```

Comment:
Why is ret initialized with 0? Seems that this would be a failure code.

Reply:
That is on purpose. From the commit message:

> Ensure `oqshybkem_init_ecp` returns an error status instead of 1 on early exit through `ON_ERR_GOTO` (ecx is unaffected).

Status code 1 is success, zero and negatives signal failures. If all goes well, a positive return value will be returned if `ret = EVP_PKEY_paramgen` below succeeds.

Previously, the `ON_ERR_GOTO(idx < 0 || idx > 6, err_init_ecp);` line below would exit this function without overwriting `ret`. As a result, `1` (success) would be returned even if a failure has occurred. Luckily there is no crash because the `evp_ctx->ctx` is also checked:
```
ON_ERR_GOTO(ret2 <= 0 || !evp_ctx->keyParam || !evp_ctx->ctx, err);
```